### PR TITLE
[IMP] web: convert core/domain_selector tests to hoot

### PR DESCRIPTION
**Pull request by juliusc2066**
task-id: 3705027

### CODE REVIEW

File: addons/web/static/tests/_framework/condition_tree_editor_test_helpers.js [hunk 1]
```diff
@@ -52,6 +52,8 @@ export const SELECTORS = {
     toggleArchive: ".form-switch",
     complexCondition: ".o_tree_editor_complex_condition",
     complexConditionInput: ".o_tree_editor_complex_condition input",
+    debugArea: ".o_domain_selector_debug_container textarea",
```
Comment (Polymorphe57) [inline addons/web/static/tests/_framework/condition_tree_editor_test_helpers.js:55]: these should not be here.
See
export const SELECTORS = {
    ...treeEditorSELECTORS,
    debugArea: ".o_domain_selector_debug_container textarea",
    resetButton: ".o_domain_selector_row > button",
};
in legacy file domain_selector_tests.js

File: addons/web/static/tests/core/domain_selector.test.js [hunk 2]
```diff
@@ -0,0 +1,2042 @@
...
+});
+
+test("reset domain", async () => {
+    await makeDomainSelector({
+        domain: `[`,
+        update(domain) {
+            expect.step(domain);
+        },
+    });
+    expect(queryOne(".o_domain_selector").textContent).toBe(
```
Comment (Polymorphe57) [inline addons/web/static/tests/core/domain_selector.test.js]: use toHaveText?

File: addons/web/static/tests/core/domain_selector.test.js [hunk 3]
```diff
@@ -0,0 +1,2042 @@
...
+    expect(SELECTORS.condition).toHaveCount(1);
+    expect(SELECTORS.addNewRule).toHaveCount(1);
+
+    await addNewRule();
+    expect(SELECTORS.condition).toHaveCount(2);
+    expect(SELECTORS.addNewRule).toHaveCount(1);
+});
+
+test("updating path should also update operator if invalid", async () => {
+    await mountWithCleanup(DomainSelector, {
```
Comment (Polymorphe57) [inline addons/web/static/tests/core/domain_selector.test.js]: why not a makeDomainSelector?

File: addons/web/static/tests/core/domain_selector.test.js [hunk 4]
```diff
@@ -0,0 +1,2042 @@
...
+    });
+    expect(getConditionText()).toBe("Foo = uid");
+});
+
+test("boolean field (readonly)", async () => {
+    const parent = await makeDomainSelector({
+        readonly: true,
+        domain: `[]`,
+    });
+    await parent.set(`[("bar", "=", True)]`);
```
Comment (Polymorphe57) [inline addons/web/static/tests/core/domain_selector.test.js]: write this with  a const toTest = [ + ...

Comment (Polymorphe57) [reply addons/web/static/tests/core/domain_selector.test.js]: ?

File: addons/web/static/tests/web_test_helpers.js [hunk 5]
```diff
@@ -29,6 +29,7 @@ import { defineModels } from "./_framework/mock_server/mock_server";
  */
 
 export { findComponent, mountWithCleanup } from "./_framework/component_test_helpers";
+export { getPickerCell, getPickerApplyButton } from "./_framework/datetime_test_helpers";
```
Comment (Polymorphe57) [inline addons/web/static/tests/web_test_helpers.js:32]: Why?

Comment (juliusc2066) [reply addons/web/static/tests/web_test_helpers.js:32]: All generic test helpers should be imported from this file and not from the _framework files

Comment (juliusc2066) [reply addons/web/static/tests/web_test_helpers.js]: but i moved it to the datetime tests folder

**Comment by Polymorphe57** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**